### PR TITLE
normalize column defaults

### PR DESCRIPTION
**Pull request by jycor**
This PR adds a new analyzer rule to normalize literal column default values.
This rule ensures that the default value is consistent for the column type (float defaults over int columns are rounded properly).
It does this by evaluating the column default, and placing that into a NewLiteral of the proper type.

Additionally, this ensures that dolt serialization receives consistent values (normalized floats and proper types).

fixes: https://github.com/dolthub/dolt/issues/8190

### CODE REVIEW

File: enginetest/queries/insert_queries.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ var InsertQueries = []WriteQueryTest{
 		WriteQuery:          "INSERT INTO mytable (s, i) VALUES ('x', '64.6');",
 		ExpectedWriteResult: []sql.Row{{types.NewOkResult(1)}},
 		SelectQuery:         "SELECT i FROM mytable WHERE s = 'x';",
-		ExpectedSelect:      []sql.Row{{int64(64)}},
+		ExpectedSelect:      []sql.Row{{int64(65)}},
```
Comment (nicktobey) [inline enginetest/queries/insert_queries.go:50]: Good catch!

File: sql/analyzer/resolve_column_defaults.go [hunk 2]
```diff
@@ -15,39 +15,38 @@
...
+//   - stripTableNamesFromDefault: column defaults headed for storage or serialization in a query result need the table
+//     names in any GetField expressions stripped out so that they serialize to strings without such table names. Table
+//     names in GetField expressions are expected in much of the rest of the analyzer, so we do this after the bulk of
+//     analyzer work.
 //
 // The `information_schema.columns` table also needs access to the default values of every column in the database, and
 // because it's a table it can't implement `sql.Expressioner` like other node types. Instead it has special handling
 // here, as well as in the `resolve_functions` rule.
-
+// TODO: rename this to resolveColumnDefaults
```
Comment (nicktobey) [inline sql/analyzer/resolve_column_defaults.go:49]: Might as well rename it before merging.

File: sql/plan/alter_default.go [hunk 3]
```diff
@@ -114,29 +114,30 @@ func (d *AlterDefaultSet) Expressions() []sql.Expression {
 	return append(transform.WrappedColumnDefaults(d.targetSchema), expression.WrapExpressions(d.Default)...)
 }
 
-func (d AlterDefaultSet) WithExpressions(exprs ...sql.Expression) (sql.Node, error) {
```
Comment (nicktobey) [inline sql/plan/alter_default.go:117]: Why were these changed to use pointer receivers?

File: sql/rowexec/show_iters.go [hunk 4]
```diff
@@ -395,7 +395,6 @@ func (i *showCreateTablesIter) produceCreateTableStatement(ctx *sql.Context, tab
 	tableCollation := table.Collation()
 	for i, col := range schema {
 		var colDefaultStr string
-		// TODO: The columns that are rendered in defaults should be backticked
```
Comment (nicktobey) [inline sql/rowexec/show_iters.go:398]: Was this comment removed because it was obsolete?

File: enginetest/queries/column_default_queries.go [hunk 5]
```diff
@@ -775,4 +775,70 @@ var ColumnDefaultTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "column default normalization: int column rounds",
```
Comment (nicktobey) [inline enginetest/queries/column_default_queries.go:779]: I think we should add additional tests for:
- An unquoted default value becomes single-quoted in the `show create` statement.
- A double quoted default value becomes single-quoted in the `show create` statement.
- An expression that wraps a literal (`i int default (1)`) is preserved and not normalized.

File: sql/analyzer/resolve_column_defaults.go [hunk 6]
```diff
@@ -413,3 +425,27 @@ func backtickDefault(wrap *expression.Wrapper) (sql.Expression, transform.TreeId
 	nd.Expr = newExpr
 	return expression.WrapExpression(&nd), transform.NewTree, nil
 }
+
+// normalizeDefault ensures that default values that are literals are normalized string literals. This is necessary for
```
Comment (nicktobey) [inline sql/analyzer/resolve_column_defaults.go:429]: So if I'm understanding this PR correctly, we're storing the normalized default as the column type, not a string literal, and it gets converted to string literal when we call `show create`. If so, we should update this comment.

**Review by nicktobey — APPROVED** [review]
Looks great! I love it.

File: sql/plan/alter_default.go (hunk 3 above)
Comment (jycor) [reply sql/plan/alter_default.go:117]: All the other methods are pointer reciever, and Goland was complaining.

File: sql/rowexec/show_iters.go (hunk 4 above)
Comment (jycor) [reply sql/rowexec/show_iters.go:398]: Yeah, we added a rule called `backtickDefaultColumnValueNames` a while ago.
